ProcessLib/RichardsMechanics/RichardsMechanicsFEM-impl.h

```diff
@@ -194,17 +227,21 @@ void RichardsMechanicsLocalAssembler<
         auto const K_LR =
             liquid_phase.property(MPL::PropertyType::bulk_modulus)
                 .template value<double>(variables, x_position, t, dt);
-        auto const porosity =
-            solid_phase.property(MPL::PropertyType::porosity)
-                .template value<double>(variables, x_position, t, dt);
+
+        // Use previous time step porosity for porosity update, ...
+        variables[static_cast<int>(MPL::Variable::porosity)] =
+            _ip_data[ip].porosity_prev;
+        auto& porosity = _ip_data[ip].porosity;
```

Comment:
Here you define the porosity and assign a value. In the next line you assign a different value. Why not like this:
```suggestion
        auto& porosity = solid_phase.property(MPL::PropertyType::porosity)
                       .template value<double>(variables, x_position, t, dt);
```

Reply:
Because the `_ip_data[ip].porosity` wouldn't be updated.
`auto& porosity` is just a reference.

Reply:
Sorry, I overlooked the reference.